ci/base-worker-image/mantic/Dockerfile

```diff
@@ -1,6 +1,6 @@
 # A minimal container for building a base worker image.
 # Buildfarm public releases are build using this image as a starting point.
-FROM ubuntu:23.04
+FROM ubuntu:mantic-20240427@sha256:565d62d2283a7cc4b3d759d9a97a5bfcebeb341166f9076a4df504f8f106cd54
```

Comment:
Why is this and the pin above for jammy desirable for this change in particular? I understand the reason to pick a non-moving target, but I'm not sure it's applicable to this PR (at least not in the same commit, without some reference in the description).

Reply:
ah sorry, should have documented it! Per the github action, it only builds this image if the `ci/base-worker-image/mantic/Dockerfile` is updated. To trigger a rebuild, I bumped the image. 
I can just sha256-pin 23.04 if you prefer.